### DISCUSSION TITLE
An addition for the scenario manager in inet

### DISCUSSION THREAD

**Discussion by anoikob** [omnetpp/omnetpp#1233] [Ideas]
As it stands (at least from what I can understand from the [documentation](https://doc.omnetpp.org/inet/api-current/neddoc/inet.common.scenario.ScenarioManager.html)) the disconnect command requires manual reconnection. Is there a possibility of adding a "reconnect" command in the future? It would be cool not to have to manually connect and specify all the necessary ports and channel types every time I want to disconnect something and then connect it again, in the exact same way.

**Comment by levy** [comment]
You can disable/enable channels instead of disconnect/reconnect. See examples/tcp_eth_reconnect/scenario2.xml for an example:

`<set-channel-param src-module="server" src-gate="ethg$o[0]" par="disabled" value="true"/>`

**Reply by anoikob** [reply]
I see, is it mechanically the same though? Does the simulator act differently depending on whether a module was disconneted vs a module's channel was disabled?

**Reply by levy** [reply]
It's not entirely the same, the modules can distinguish between disconnecting or disabling a connection. It also looks different in the user interface, because the connection is still there, it's just disabled. But in both cases the packets can't be sent through the connection. So depending on what you want to do, it may behave the same way from your perspective.

**Reply by anoikob** [reply]
Ok i see, thanks for the clarification
